### PR TITLE
Support gcp.source_location being set as other types

### PR DESCRIPTION
**Pull request by braydonk**
By copying the logic from parsing gcp.http_request, this PR adds support to also parse `gcp.source_location` from types other than Bytes.

This fixes the basic case of #792 such that the collector does not panic when setting `gcp.source_location` as something other than bytes.

### CODE REVIEW

File: exporter/collector/logs.go
Comment (quentinmit) [inline exporter/collector/logs.go:396]: Maybe worth refactoring this into a common function that's used by both call sites? (Especially since it's going to be needed a third time to support `operation`)

File: exporter/collector/logs_test.go
Comment (quentinmit) [inline exporter/collector/logs_test.go:399]: What makes this test "bad"?

File: exporter/collector/logs.go
Comment (braydonk) [reply exporter/collector/logs.go]: Done in [a0f37b2](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/794/commits/a0f37b2a6901f8c2529db678a696f414e0a6bf5b)

File: exporter/collector/logs_test.go
Comment (braydonk) [reply exporter/collector/logs_test.go]: It's because the value provided for `gcp.source_location` can't actually be parsed, the json Unmarshal fails due to an invalid type in one of the fields. Is there a nicer way I should say that concisely in the test or should I add a comment?

File: exporter/collector/logs.go
Comment (dashpole) [inline exporter/collector/logs.go:127]: Does this need to be public?  Same question for other error below.

Comment (braydonk) [reply exporter/collector/logs.go:127]: I tend to make these public out of habit so that they can be used in test assertions, but since we don't use the `_test` package for these tests I can make these private.

Comment (braydonk) [reply exporter/collector/logs.go]: Done in [4b7052e](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/794/commits/4b7052e0d2962cb46c613a4f1e56d431b7ad894e)

**Review by dashpole — APPROVED** [review]

**Comment by dashpole** [comment]
@quentinmit i'll wait for your approval to merge

File: exporter/collector/logs.go
Comment (quentinmit) [inline exporter/collector/logs.go:656]: Why not put the `json.Unmarshal` in the function too? Then the signature can just be `unmarshalAttribute(v pcommon.Value, out any) error`

**Review by quentinmit — APPROVED** [review]
See comment, but otherwise this LGTM

File: exporter/collector/logs_test.go
Comment (quentinmit) [reply exporter/collector/logs_test.go]: `"log with wrong type in sourceLocation (bytes)"`, perhaps?

File: exporter/collector/logs.go
Comment (braydonk) [reply exporter/collector/logs.go]: Done in [992e990](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/794/commits/992e99041c6f46520f344b0da0cb006b98b972ad)

**Comment by braydonk** [comment]
Let's see if the e2e tests will work today :pray:

**Comment by braydonk** [comment]
I still don't seem to have permission to run these :(

**Pull request merged**
